### PR TITLE
#39 Implement Sign in page design

### LINKED ISSUE #39: Implement Sign In page design
Acceptance criteria:

Sign In page design is implemented with button hover states as per design.

### PR DESCRIPTION
**Pull request by fola-ooo**
Fixes #39

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,11 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@fortawesome/fontawesome-free": "^5.15.1",
+    "@fortawesome/fontawesome-svg-core": "^1.2.32",
+    "@fortawesome/free-brands-svg-icons": "^5.15.1",
```
Comment (toystars) [inline package.json:8]: @FolaDee `react-icons` already contain `font-awesome`, no?

File: src/modules/sign/components/Sign.jsx [hunk 2]
```diff
@@ -1,7 +1 @@
...
-
-export default Sign;
+import React from "react";

function Sign() {

	return <div></div>;
}

export default Sign;
```
Comment (toystars) [inline src/modules/sign/components/Sign.jsx:1]: Why is the code mangled? Is this a formatting issue?

**Review by toystars — CHANGES_REQUESTED** [review]
Please format your JavaScript files correctly; then I can review. The code is unreadable.

Also, why is the PR titled: `First commit`?

File: package.json (hunk 1 above)
Comment (fola-ooo) [reply package.json:8]: @toystars I will remove it

File: src/modules/sign/components/Sign.jsx (hunk 2 above)
Comment (fola-ooo) [reply src/modules/sign/components/Sign.jsx:1]: @toystars probably a formatting issue, I will check that also

**Review by toystars — COMMENTED** [review]
@FolaDee as you can see from this screenshot, the code is unreadable. Which IDE do you use?

Screenshot 2020-12-08 at 10 07 52 AM

**Comment by fola-ooo** [comment]
> @FolaDee as you can see from this screenshot, the code is unreadable. Which IDE do you use?
> 
> Screenshot 2020-12-08 at 10 07 52 AM

@toystars I sincerely apologize for this. I should have checked thoroughly before asking for a review. It was an encoding problem in my IDE which I have fixed.

File: src/modules/sign/components/Signin.css [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+.svg_amazon {
+  position: absolute;
+  width: 100px;
+  height: 38px;
+  left: 27rem;
+  top: 745px;
+}
+.svg_airbnb {
+  position: absolute;
+  width: 100px;
```
Comment (toystars) [inline src/modules/sign/components/Signin.css:69]: Please switch from using `px` and use `rem` as your unit of measurements.

File: src/modules/sign/components/Signin.css [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+}
+.svg_netflix {
+  position: absolute;
+  width: 100px;
+  height: 20px;
+  left: 45rem;
+  top: 752px;
+}
+.svg_microsoft {
+  position: absolute;
```
Comment (toystars) [inline src/modules/sign/components/Signin.css:89]: I see lots of absolute positioning here. I suppose this is a direct translation of the Figma styles to CSS. This is not ideal and I propose you use flexbox to position your elements. This also allows for better responsiveness.

File: src/modules/sign/components/Signin.jsx [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import AirbnbLogo from "../icons/airbnblogo";
+import UberLogo from "../icons/uberlogo";
+import MicrosoftLogo from "../icons/mslogo";
+import NetflixLogo from "../icons/netflixlogo";
+import GitSignIn from "./gitSignIn";
+import "./Signin.css";
+
+const SignIn = () => {
+  return (
+    <div>
```
Comment (toystars) [inline src/modules/sign/components/Signin.jsx:13]: Nitpicking here, but indentation could be improved.

File: src/modules/sign/components/gitSignIn.jsx [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    : {};
+  const gitlabActiveStyles = gitlabActive
+    ? {
+      border: "1px solid black",
+    }
+    : {};
+  useEffect((e) => {
+    console.log(e);
+    const githubButton = document.getElementById("button");
```
Comment (toystars) [inline src/modules/sign/components/gitSignIn.jsx]: Please avoid using vanilla JavaScript ways of getting reference to DOM elements. If you need a reference, you can use [React Ref](https://reactjs.org/docs/refs-and-the-dom.html).

File: src/modules/sign/components/gitSignIn.jsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    githubButton.addEventListener("mouseover", function github () { githubSetActive(true) });
+    githubButton.addEventListener("mouseout", function github () { githubSetActive(false) });
+    const gitlabButton = document.getElementById("button2");
+    gitlabButton.addEventListener("mouseover", function gitlab () { gitlabSetActive(true) });
```
Comment (toystars) [inline src/modules/sign/components/gitSignIn.jsx]: This approach works, but it is not elegant and also not the react way of adding event listeners. You could do something as below

```js

const mouseOverFunction = () => {
    githubSetActive(true);
}

const mouseOutFunction = () => {
    githubSetActive(false);
}

<button style={githubActiveStyles} className="button" onMouseOver={mouseOverFunction} onMouseOut={mouseOutFunction}>
    <img src={githubActive ? activateGithub : github} alt="" />
    <span className="button-text">Continue with GitHub</span>
</button>
```

With this, you won't have to bother about getting element by id or removing the event listeners. React will automatically remove the event listeners when the component unmounts. You also won't need the `useEffect` hook.

File: src/modules/sign/components/gitSignIn.jsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <div className="git">
+      <button id="button" style={githubActiveStyles} className="button">
+        <img src={githubActive ? activateGithub : github} alt="" />
+        <span className="button-text">Continue with GitHub</span>
+      </button>
+
+      <br/>
+
+      <button id="button2" style={gitlabActiveStyles} className="button2">
```
Comment (toystars) [inline src/modules/sign/components/gitSignIn.jsx]: Please ensure to give CSS classes more descriptive names than `button2`. This makes it easier when someone else reads your code.

**Review by toystars — CHANGES_REQUESTED** [review]

**Review by toystars — APPROVED** [review]

**Pull request merged**